sklearn/utils/_array_api.py

```diff
@@ -86,6 +86,22 @@ def unique_values(self, x):
     def concat(self, arrays, *, axis=None):
         return numpy.concatenate(arrays, axis=axis)
 
+    def reshape(self, x, shape, *, copy=None):
+        """Gives a new shape to an array without changing its data.
+
+        The Array API specification requires shape to be a tuple.
+        https://data-apis.org/array-api/latest/API_specification/generated/array_api.reshape.html
+        """
+        if not isinstance(shape, tuple):
+            raise TypeError("shape must be a tuple")
+
+        if copy is True:
+            x = x.copy()
+        elif copy is False:
+            x.shape = shape
+            return x
+        return numpy.reshape(x, shape)
```

Comment:
Is this always valid? I just tried with NumPy with a shape with a `-1` and it works but why not call `numpy.reshape(x, shape)` as usual?


```suggestion
        if copy is True:
            x = x.copy()
        return numpy.reshape(x, shape)
```

Reply:
Yea, I think calling `numpy.reshape` is fine for our use case.